sdk/go/common/workspace/plugins.go

```diff
@@ -2218,7 +2243,7 @@ func getPluginInfoAndPath(
 	// plugins are not, or has simply set IGNORE_AMBIENT_PLUGINS. So, if possible, look next to the instance
 	// of `pulumi` that is running to find this bundled plugin.
 	var bundledPath string
-	if IsPluginBundled(kind, name) {
+	if IsPluginBundled(spec.Kind, spec.Name) {
```

Comment:
Not necessarily for this PR but I did a code search when I was looking at this work and this could safely be changed to take a spec too I think.